src/FakeItEasy/Core/FakeManager.CancellationRule.cs

```diff
@@ -83,7 +83,8 @@ private static bool IsValueTask(Type type, out Type? valueType)
 
                 if (type.IsGenericType &&
                     !type.IsGenericTypeDefinition &&
-                    type.FullName.StartsWith("System.Threading.Tasks.ValueTask`", StringComparison.Ordinal))
+                    type.FullName is string fullName &&
+                    fullName.StartsWith("System.Threading.Tasks.ValueTask`", StringComparison.Ordinal))

```

Comment:
FullName can be null in [some situations](https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname?view=net-5.0)